light-client/src/fork_detector.rs

```diff
@@ -101,22 +103,24 @@ impl ForkDetector for ProdForkDetector {
                 continue;
             }
 
-            state.light_store.update(&trusted_state, Status::Verified);
+            // XXX: Shouldn't this have already happened? Why do we need
+            // to re-mark the trusted block as verified?
```

Comment:
cc @romac

Reply:
Yes, I believe that to be the case:
https://github.com/informalsystems/tendermint-rs/blob/d498fe2d5518a8887f501006a6b477f24621b5cc/light-client/src/light_client.rs#L203-L206

Fork detection on this particular light block will only occur on verdict successful so it should be in the store marked as verified.

This has come up before by I just drop it again here as it seems relevant. Sharing mutable access to the light store between components feels a bit cumbersome. It might be better in future iterations have a single owner (probably the supervisor) to ensure consistent mutation.

Reply:
lol. I've asked the same question a few weeks ago. https://github.com/informalsystems/tendermint-rs/pull/287#discussion_r438590326

Reply:
The ‘state’ here is a fresh in-memory state, not the primar’s or any of the witnesses’ own states, so we need to insert the trusted block to have a trusted state to work from for verification. I guess the mistake I made here is to call ‘update’ instead of ‘insert’, which in this case are equivalent but does not communicate the proper intent. Will fix!

Reply:
Fixed in https://github.com/informalsystems/tendermint-rs/pull/404/commits/b45851877f695da5e162713b5ecc50d2d5feef55